BGL/include/CGAL/boost/graph/selection.h

```diff
@@ -521,6 +522,100 @@ reduce_vertex_selection(
   return out;
 }
 
+/**
+ * \ingroup PkgBGLSelectionFct
+ *
+ * Expands a selection of faces so that their removal does not create non manifold vertex.
+ *
+ * @tparam TriangleMesh a model of `FaceGraph` that is triangulated.
+ * @tparam FaceRange a range of `boost::graph_traits<TriangleMesh>::%face_descriptor`,
+ * with an iterator type model of `ForwardIterator`.
+ * @tparam  IsSelectedMap a model of `ReadWritePropertyMap` with
+ * `boost::graph_traits<TriangleMesh>::%face_descriptor` as key and `bool` as value.
+
+ * @param tm the triangle mesh.
+ * @param faces_to_be_deleted the range of selected faces.
+ * @param is_selected a property map containing the selected-or-not status of each face of `tm`.
+ * It must associate `true` to each face of `faces_to_be_deleted` and `false` to any other face of `tm`.
+ * It will be modified if the face selection must be expanded.
+ *
+ **/
+template<class TriangleMesh, class FaceRange, class IsSelectedMap>
+void expand_face_selection_for_removal(TriangleMesh& tm,
+                                       const FaceRange& faces_to_be_deleted,
+                                       IsSelectedMap is_selected)
+{
+  typedef typename boost::graph_traits<TriangleMesh>::vertex_descriptor vertex_descriptor;
+  typedef typename boost::graph_traits<TriangleMesh>::face_descriptor face_descriptor;
+  typedef typename boost::graph_traits<TriangleMesh>::halfedge_descriptor halfedge_descriptor;
+
+  boost::unordered_set<vertex_descriptor> vertices_to_be_deleted;
```

Comment:
I think the name is misleading because vertices on the boundary of the selection will not be deleted.